src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -204,4 +205,26 @@ trait RVD {
   def sample(withReplacement: Boolean, p: Double, seed: Long): RVD
 
   def write(path: String, codecSpec: CodecSpec): Array[Long]
+
+  def shuffle(typ: OrderedRVDType, partitioner: OrderedRVDPartitioner): OrderedRVD = {
+    assert(typ.rowType == rowType)
+    OrderedRVD.shuffle(typ, partitioner, rdd)
+  }
+
+  def assertOrdered(typ: OrderedRVDType, partitioner: OrderedRVDPartitioner): OrderedRVD = {
```

Comment:
The other way to write this is a constructor OrderedRVD(RVD, ...).

Reply:
I kind of prefer this because it's more scalable in RVD subclasses.

Reply:
What do you mean by it's more scalable?

Reply:
I mean, it seems bad to have methods in the base class for each derived class.

Reply:
In #3159 I added `RVD.constrainToOrderedPartitioner`, which has some overlap with this. It is implemented in `UnpartitionedRVD` by `shuffle`, and in `OrderedRVD` by a narrow-dependency repartition I wrote. It's slightly lower level than this, because it is meant to be used in a case where you want an `OrderedRVD` with *exactly* the given partitioner, dropping any data that falls outside the new partitions.